### ISSUE TITLE
Freqtrade closes trade below candlestick low price

### ISSUE THREAD

**Issue by antonino-git** [freqtrade/freqtrade#5816]
## Describe your environment

  * Operating system: Ubuntu 16.04
  * Python Version: 3.7.5
  * CCXT version: 1.57.3
  * Freqtrade Version: 2021.9
  
Note: All issues other than enhancement requests will be closed without further comment if the above template is deleted or not filled out.

## Describe the problem:

*Explain the problem you have encountered*

I created a simple strategy that works on the 5 min timeframe. With this strategy the bot buys when the RSI <= 25 and sells when the price closes at least 2% below the trailing stoploss. 

When I backtest the strategy I notice that the bot closes most of the trades after 0 minutes at a price that is below the candlestick low. Plotting the dataframe helped to visualize the issue. 

### Steps to reproduce:

  1. Run the backtest using Binance historical data: freqtrade backtesting --strategy RsiTestStrategy --timerange=20210121-20210122 -p CTK/USDT -c config_backtrader.json
  2. Check the first trade in the backtest log
  3. Plotting helps to visualize the issue: freqtrade plot-dataframe --strategy RsiTestStrategy -p CTK/USDT --timerange=20210121-20210122 --indicators2 rsi  --config config_backtrader.json
  
### Observed Results:

  * What happened?
       The bot opens a trade on the CTK/USDT pair the 21/Jan/2021 at 14:10 purchasing CTK at the price of 1.1754 USDT, setting a training stoploss at 1.151892. 

       Candle details:
        Open: 1.1754
        Close: 1.1586
        High: 1.2049
        Low: 1.1586

       After 0 minutes the bot triggers the stoploss and closes the position selling the trade at 1.151892 (Value below the candlestick low).   
 
       The same problem repeats for most of the trades 

  * What did you expect to happen?
     The bot shouldn't have closed the trade yet
      I'm not sure this is a bug, I probably messed up with the configuration. 
      
       Thanks for your help.

### Relevant code exceptions or logs

Note: Please copy/paste text of the messages, no screenshots of logs please.

  ```
  **The trade log:** 
{"pair":"CTK/USDT","stake_amount":    100.0,"amount":85.07742045,"open_date":"2021-01-21 14:10:00+00:00",
"close_date":"2021-01-21 14:10:00+00:00","open_rate":1.1754,"close_rate":1.151892,"fee_open":0.001,
"fee_close":0.001,"trade_duration":0,"profit_ratio":-0.02195804,"profit_abs":-2.198,
"sell_reason":"trailing_stop_loss","initial_stop_loss_abs":1.151892,"initial_stop_loss_ratio":-0.02,
"stop_loss_abs":1.1808020000000001,"stop_loss_ratio":-0.02,"min_rate":1.1586,"max_rate":1.2049,
"is_open":false,"buy_tag":null,"open_timestamp":1611238200000.0,"close_timestamp":1611238200000.0}
```

``` python
class RsiTestStrategy(IStrategy):
   
    INTERFACE_VERSION = 2

    minimal_roi = {
        "0": 100
    }

    stoploss = -0.02

    trailing_stop = True

    timeframe = '5m'

    process_only_new_candles = False

    use_sell_signal = True
    sell_profit_only = False
    ignore_roi_if_buy_signal = False

    startup_candle_count: int = 14

    order_types = {
        'buy': 'limit',
        'sell': 'limit',
        'stoploss': 'market',
        'stoploss_on_exchange': False
    }

    order_time_in_force = {
        'buy': 'gtc',
        'sell': 'gtc'
    }
    
    plot_config = {
        # Main plot indicators (Moving averages, ...)
        'main_plot': {
            'tema': {},
            'sar': {'color': 'white'},
        },
        'subplots': {
            # Subplots - each dict defines one additional plot
            # "MACD": {
            #     'macd': {'color': 'blue'},
            #     'macdsignal': {'color': 'orange'},
            # },
            "RSI": {
                'rsi': {'color': 'red'},
            }
        }
    }
    def informative_pairs(self):
        return []

    def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        # RSI
        dataframe['rsi'] = ta.RSI(dataframe)

        return dataframe

    def populate_buy_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
       
        dataframe.loc[
            (
                (dataframe['rsi'] <= 25) &  # Guard: tema below BB middle
                (dataframe['volume'] > 0)  # Make sure Volume is not 0
            ),
            'buy'] = 1

        return dataframe

    def populate_sell_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
       
        return dataframe
```

```
**The configuration I use:** 
{
    "max_open_trades": 10,
    "stake_currency": "USDT",
    "stake_amount": "unlimited",
    "available_capital":1000,
    "tradable_balance_ratio": 0.99,
    "fiat_display_currency": "USD",
    "timeframe": "5m",
    "dry_run": true,
    "cancel_open_orders_on_exit": false,
    "unfilledtimeout": {
        "buy": 10,
        "sell": 30,
        "unit": "minutes"
    },
    "bid_strategy": {
        "price_side": "bid",
        "ask_last_balance": 0.0,
        "use_order_book": true,
        "order_book_top": 1,
        "check_depth_of_market": {
            "enabled": false,
            "bids_to_ask_delta": 1
        }
    },
    "ask_strategy": {
        "price_side": "ask",
        "use_order_book": true,
        "order_book_top": 1
    },
    "exchange": {
        "name": "binance",
        "key": "",
        "secret": "",
        "skip_pair_validation": true,
        "ccxt_config": {"enableRateLimit": true},
        "ccxt_async_config": {
            "enableRateLimit": true,
            "rateLimit": 200
        },
        "pair_whitelist": [
            ".*/USDT" 
        ],
        "pair_blacklist": [
            "BNB/.*",
            "BUSD/.*",
            "USDC/.*",
            "EUR/.*",
            ".*UP/.*",
            ".*BULL/.*",
            ".*DOWN/.*",
            ".*BEAR/.*",
            "TUSD/.*",
            "USDT/.*",
            "USDP/.*",
            "LAZIO/.*",
            "JUV/.*",
            "FIO/.*",
            "BAR/.*",
            "ACM/.*",
            "PSG/.*",
            "ATM/.*"
        ]
    },
    "pairlists": [
        {
            "method": "StaticPairList",
            "allow_inactive": true
        }
    ],
    "edge": {
        "enabled": false,
        "process_throttle_secs": 3600,
        "calculate_since_number_of_days": 7,
        "allowed_risk": 0.01,
        "stoploss_range_min": -0.01,
        "stoploss_range_max": -0.1,
        "stoploss_range_step": -0.01,
        "minimum_winrate": 0.60,
        "minimum_expectancy": 0.20,
        "min_trade_number": 10,
        "max_trade_duration_minute": 1440,
        "remove_pumps": false
    },
    "telegram": {
        "enabled": false,
        "token": "",
        "chat_id": ""
    },
    "bot_name": "freqtrade",
    "initial_state": "running",
    "forcebuy_enable": false,
    "internals": {
        "process_throttle_secs": 5
    }
}

```

![error2](https://user-images.githubusercontent.com/65393791/139533104-0c0d203f-4287-408a-ac03-525b9a38bb2b.png)
2

**Comment by antonino-git** [comment]
![Error_plot](https://user-images.githubusercontent.com/65393791/139533135-f9f05a7e-8a22-4c3e-b337-1830e896956c.png)

**Comment by xmatthias** [comment]
That's a pretty nice, minimal reproduction of an issue - this helps a lot in quickly finding the problem :+1: 

We try hard to assume "worst case" for trailing stops that happen on the opening candle.
Therefore, we assume price moves 1-2 ticks above open - and then drops down (it could also move to high and then drop down ... we simply don't know).

Obviously, we should be limiting the sell to stay within the candle - which we do towards the upper side, but miss on the lower side.
the fix should be quick - but in the end, it's really only showing that the stoploss was trailing too tight - causing an immediate stop at the opening candle.

**Comment by antonino-git** [comment]
Glad this was helpful. Thanks for the great work you are doing in maintaining this amazing bot.

**Comment by antonino-git** [comment]
@xmatthias, the PR didn't fix the issue. The bot closes all the trades despite the price. In the image below you can see that the bot closes the trade even if the price is only 0.3% below the maximum price. In the strategy, the stoploss is set 2% below the trailing stoploss 

![Error_plot2](https://user-images.githubusercontent.com/65393791/139538487-39f6bbaa-0547-4c2c-b41d-a744a713efb9.png)

**Comment by xmatthias** [comment]
Please familiarize yourself with the [backtesting assumptions](https://www.freqtrade.io/en/stable/backtesting/#assumptions-made-by-backtesting), specifically with the segment about trailing stoploss.

As you're using trailing stoploss without any conditions, you'll hit a stoploss whenever the difference between high and low is > your trailing stop value (assuming you're out of the trade, before moves up in future candles).

So while your trade shows a sell at -0.3% - it'll be 2% below this candle's high (assuming in this case the buy happened at the same candle, otherwise 2% below the highest high within this trade's range).

**Comment by antonino-git** [comment]
@xmatthias, sorry to bother you again but I'm still not sure about this stop loss.

Look at the candle I underlined in the picture:
Open: 0.99
High: 1.0477
Low: 0.9822
Close: 0.9983

The bot opens a trade using the open price 0.99 and sets the stoploss at 0.9702
According to the documentation, the bot evaluates first the candle high price and updates the stoploss. Now the stoploss should be max price - 0.02% = 1.0477 - 0.020954 = 1.026746. 
The bot should evaluate the candle low price (0.9822), since this is below the stop loss, the bot should close the trade using the stoploss price(According to the documentation too). But for some reason, it closes the trade using the candle low price. 
Am I missing something? 

```
**Trade log:** 

{"pair"    :"CTK/USDT","stake_amount":97.71500482,"amount":98.70202507,"open_date":"2021-01-21 15:25:00+00:00",
"close_date":"2021-01-21 15:25:00+00:00","open_rate":0.99,"close_rate":0.9822000000000001,"fee_open":0.001,
"fee_close":0.001,"trade_duration":0,"profit_ratio":-0.00986105,"profit_abs":-0.96453593,"sell_reason":"trailing_stop_loss",
"initial_stop_loss_abs":0.9702,"initial_stop_loss_ratio":-0.02,"stop_loss_abs":1.0267460000000002,"stop_loss_ratio":-0.02,
"min_rate":0.9822000000000001,"max_rate":1.0477,"is_open":false,"buy_tag":null,"open_timestamp":1611242700000.0,
"close_timestamp":1611242700000.0},
```
![error4](https://user-images.githubusercontent.com/65393791/139541474-01a063e7-de4a-410d-82d1-a38c872837c9.png)

**Comment by xmatthias** [comment]
for opening candles (where the trade is entered) that trigger a trailing-stop - the sell is assuming the "worst possible case" - which is "open -> "slightly above open -> low" (never even reaching high).

Otherwise it's very easy to create a strategy that exploits the difference between open to high and assumes trailing-stop will hit at "high * 0.98" - which is simply unrealistic (the assumption is that intra-candle movements are not known).

Obviously you can still exploit this after the first candle - but it's less likely, and we've seen a lot less "exploiting" strategies since this change.

**Comment by antonino-git** [comment]
but the "worst possible case" should not trigger the stop loss. 
Trade: Open[buy: 0.99, Stoploss: 0.9702] -> Low[current price: 0.9822, Stoploss: 0.9702] The stoploss is below the low

**Comment by xmatthias** [comment]
yes it does - it assumes price raises just enough to move stoploss above the low (which it does - as high is > 2% above low).

**Comment by antonino-git** [comment]
ok, I got it. This simulates the very worst scenario which is different from the average case, but I guess it is hard to properly simulate the average case.  

In this case, a few sentences in the documentation in https://www.freqtrade.io/en/stable/backtesting/#assumptions-made-by-backtesting would need to be updated:

"Trailing stoploss: High happens first - adjusting stoploss"

**Comment by xmatthias** [comment]
I'm not sure about that - we do assume that high happens before first ... so this is not wrong.
It's just that the adjustment for the initial candle is different ... but that's a detail a good strategy (one not trying to exploit the backtesting method) should not have to care about.
